Homework 2

er3mey opened the pull request.

er3mey committed:
homework_2

er3mey committed:
Merge branch 'master' of https://github.com/er3mey/reboot_java into homework_2

kataus commented on lesson_3_DiyList/src/main/java/ru/sbrf/MyOwnArrayList.java:
```diff
@@ -0,0 +1,235 @@
+        System.out.println("]");
+    }
+
+    E elementData(int index) {
```
default - довольно опасный модификатор видимости. Не стоит использовать его, если не можете твердо мотивировать зачем

er3mey replied:
> default - довольно опасный модификатор видимости. Не стоит использовать его, если не можете твердо мотивировать зачем

Учел, теперь использую модификатор private

kataus commented on lesson_3_DiyList/src/main/java/ru/sbrf/MyOwnArrayList.java:
```diff
@@ -0,0 +1,235 @@
+    }
+
+    private void grow() {
+        int newSize = elementData.length + 1;
```
Расширять каждый раз на 1 элемент массив - уж очень дорого. Это нужно выделить память, скопировать, добавляем работы GC. В реальности расширение происходит скачкообразно. Пример можно подсмотреть в референском ArrayList

er3mey replied:
> Расширять каждый раз на 1 элемент массив - уж очень дорого. Это нужно выделить память, скопировать, добавляем работы GC. В реальности расширение происходит скачкообразно. Пример можно подсмотреть в референском ArrayList

Поправил, теперь массив расширяется более экономично (size^2)

er3mey committed:
homework_2

elementData теперь с модификатором private
размер массива теперь меняется более экономично (size^2)
